### PR TITLE
BHBC-1210

### PR DESCRIPTION
**Pull request by cgarrettjones**
# Overview

## Links to jira tickets

- [BHBC-1210](https://quartech.atlassian.net/browse/BHBC-1210)

## This PR contains the following changes

      - added model entities data_package, occurrence_submission_data_package and system_metadata_constant
      - populating initial values for system metadata populate_system_constants
      - folded permits to system user relationship into model pipeline
      - adjusted smoke test to delete created records

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: database/src/migrations/release.0.21/api_delete_survey.sql [hunk 1]
```diff
@@ -29,12 +29,11 @@ begin
...
   delete from study_species where survey_id = p_survey_id;
   delete from survey_funding_source where survey_id = p_survey_id;
-  delete from block_observation where survey_id = p_survey_id;
 
   update permit set survey_id = null where survey_id = p_survey_id;
 
 exception
   when others THEN
-    raise;
+    raise;
```
Comment (sdevalapurkar) [inline database/src/migrations/release.0.21/api_delete_survey.sql:37]: super nit:
```suggestion
    raise;
```

File: database/src/migrations/release.0.21/api_delete_survey.sql [hunk 2]
```diff
@@ -29,12 +29,11 @@ begin
...
 
   update permit set survey_id = null where survey_id = p_survey_id;
 
 exception
   when others THEN
-    raise;
+    raise;    
 end;
-$$;
+$$;
```
Comment (sdevalapurkar) [inline database/src/migrations/release.0.21/api_delete_survey.sql:39]: nit:
```suggestion
$$;

```

**Review by sdevalapurkar — APPROVED** [review]
code looks good to me, a few nits

**Review by anissa-agahchen — APPROVED** [review]
@cgarrettjones, I ran your PR locally, and all is well.  Would you like to merge it into dev?  It requires dropping the current DB first.  Let me know if you want me to help.

**Pull request merged**